libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx

```diff
@@ -313,8 +313,29 @@ export const ContractSelection = (props: ContractSelectionProps) => {
 
           const { data: scanData } = await axios.post('https://solidityscan.remixproject.org/downloadResult', { url })
           const scanReport: ScanReport = scanData.scan_report
-
           if (scanReport?.multi_file_scan_details?.length) {
+            for (const template of scanReport.multi_file_scan_details) {
+              if (template.metric_wise_aggregated_findings?.length) {
+                const { metric_wise_aggregated_findings } = template
+                const positions = []
+                for (const details of metric_wise_aggregated_findings) {
+                  const { findings } = details
+                  for (const f of findings){
+                    const position = {
+                      start: {
+                        line: f.line_nos_start[0] - 1,
+                      },
+                      end: {
+                        line: f.line_nos_end[0],
+                      }
+                    }
+                    await plugin.call('editor', 'highlight', position)
```

Comment:
at which point does this get unhighlighted?

Reply:
I have just added a commit to discard highlighting on multiple actions

Reply:
highlighting is removed